controllers/datadogagent/clusteragent.go

```diff
@@ -178,6 +178,10 @@ func newClusterAgentDeploymentFromInstance(logger logr.Logger, dda *datadoghqv1a
 	}
 
 	annotations := getDefaultAnnotations(dda)
+	for key, val := range dda.Annotations {
```

Comment:
I'm not sure we want to add dda.Annotations to the child deployment

@vboulineau, what is you opinion here? since I know that you have worked on this topic recently.

Reply:
We should not do it due to: https://github.com/DataDog/datadog-operator/pull/285

Reply:
okay thanks and good catch. I initially added that because it was panic'ing, but as I rebased I included your change and missed the rational why we removed them 👍